Library/RSBot.Core/Objects/Position.cs

```diff
@@ -3,125 +3,225 @@
+                _XSector = (byte)(value & 0xFF);
             }
         }
-
         /// <summary>
-        /// Gets or set the regional Y sector.
+        /// Gets a value indicating whether this instance is in dungeon.
         /// </summary>
-        private byte _YSector;
-
-        public byte YSector
-        {
-            get => !IsInDungeon ? ((WorldYOffset == 0) ? _YSector : (byte)(WorldYOffset / 1920)) : _YSector;
-            set
-            {
-                _YSector = value;
-            }
-        }
-
+        /// <value><c>true</c> if this instance is in dungeon; otherwise, <c>false</c>.</value>
+        public bool IsInDungeon => _RegionID > short.MaxValue;
         /// <summary>
-        /// Gets or set the regional XOffset
+        /// Gets the regional X sector.
         /// </summary>
-        public float WorldXOffset;
-
+        public byte XSector { get => _XSector; }
         /// <summary>
-        /// Gets or set the regional YOffset
+        /// Gets the regional Y sector.
         /// </summary>
-        public float WorldYOffset;
-
+        public byte YSector { get => _YSector; }
         /// <summary>
-        /// Gets or sets the x offset.
+        /// Gets or set the position X from map
         /// </summary>
         public float XOffset
         {
-            get => IsInDungeon ? XOffsetFromX(XCoordinate) : WorldXOffset % 1920;
+            get { return _XOffset; }
             set
             {
-                WorldXOffset = _XSector * 1920 + value;
+                if (IsInDungeon)
+                {
+                    _XOffset = value;
+                    _XCoordinate = value / 10;
+                    _XSector = (byte)(((128.0 * 192.0 + (128 * 192 + value / 10)) / 192.0) - 128);
+                }
+                else
+                {
+                    _XOffset = value;
+                    while (_XOffset < 0)
+                    {
+                        _XOffset += 1920;
+                        _XSector -= 1;
+                    }
+                    while (_XOffset > 1920)
+                    {
+                        _XOffset -= 1920;
+                        _XSector += 1;
+                    }
+                    _XCoordinate = (_XSector - 135) * 192 + value / 10;
+                }
             }
         }
-
         /// <summary>
-        /// Gets or sets the z offset.
-        /// </summary>
-        public float ZOffset;
-
-        /// <summary>
-        /// Gets or sets the y offset.
+        /// Gets or set the position Y from map.
         /// </summary>
         public float YOffset
         {
-            get => IsInDungeon ? YOffsetFromY(YCoordinate) : WorldYOffset % 1920;
+            get { return _YOffset; }
             set
             {
-                WorldYOffset = _YSector * 1920 + value;
+                if (IsInDungeon)
+                {
+                    _YOffset = value;
+                    _YCoordinate = value / 10;
```

Comment:
Do not always enter data in the X Coordinate. Calculate only when necessary and return the result.

Just example:
```
public float YCoordinate => IsInDungeon ? YOffset / 10.0f : (_YSector - 92) * 192 + YOffset/ 10.0f;
```